File: system/sw/BECS_Lib.cs
// Copyright 2015 Craig Welch
//
// Licensed under the MIT license. See LICENSE.txt file in the project root 
// for full license information.

namespace be {

public class BECS_Lib {
  
    public static void putCallId(string name, int iid) {
        BECS_Ids.callIds[name] = iid;
        BECS_Ids.idCalls[iid] = name;
    }
    
    public static int getCallId(string name) {
        return BECS_Ids.callIds[name];
    }
    
    public static void putNlcSourceMap(string clname, int[] vals) {
      BECS_Runtime.smnlcs.Add(clname, vals);
    }
    
    public static void putNlecSourceMap(string clname, int[] vals) {
      BECS_Runtime.smnlecs.Add(clname, vals);  
    }
    
}

}

File: system/sw/BECS_Lib.swift
// Copyright 2015 The Brace Authors. All rights reserved.
// Use of this source code is governed by the BSD-3-Clause
// license that can be found in the LICENSE file.

class BECS_Lib {
  
    static func putCallId(name:String?, iid:Int?) {
        //BECS_Ids.callIds[name] = iid;
        //BECS_Ids.idCalls[iid] = name;
    }
    
    static func getCallId(name:String?) -> Int? {
        //return BECS_Ids.callIds[name];
        return nil;
    }
    
    static func putNlcSourceMap(clname:String?, vals:[Int]?) {
      //BECS_Runtime.smnlcs.Add(clname, vals);
    }
    
    static func putNlecSourceMap(clname:String?, vals:[Int]?) {
      //BECS_Runtime.smnlecs.Add(clname, vals);  
    }
    
}
